gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/ManifestBasedDataset.java

```diff
@@ -124,29 +131,15 @@ public Iterator<FileSet<CopyEntity>> getFileSetIterator(FileSystem targetFs, Cop
   }
 
   private static boolean shouldCopy(FileStatus fileInSource, FileStatus fileInTarget, CopyConfiguration copyConfiguration) {
-    if (fileInSource.isDirectory()) {
-      // If dir is specified in the manifest, we blindly syn them on the metadata without concerning the mod time
-      return true;
+    if (fileInSource.isDirectory() || fileInSource.getModificationTime() == fileInTarget.getModificationTime()) {
+      // if source is dir or source and dst has same version, we compare the permission to determine whether it needs another sync
+      OwnerAndPermission replicatedPermission = CopyableFile.resolveReplicatedOwnerAndPermission(fileInSource, copyConfiguration);
+      return !replicatedPermission.isHavingSameOwnerAndPermission(fileInTarget);
     }
     if (fileInSource.getModificationTime() > fileInTarget.getModificationTime()) {
```

Comment:
should be equivalent to replace `if` with `return` and leave off the rest of the method...